src/main/webapp/i18n/en/programmingExercise.json

```diff
@@ -85,7 +85,13 @@
                 "weightCouldNotBeUpdated": "The weights for the following test cases could not be updated: {{testCases}}.",
                 "saveTestCases": "Save test cases",
                 "testCasesUpdated": "The test cases were updated successfully.",
-                "testCasesCouldNotBeUpdated": "The following test cases could not be updated: {{testCases}}."
+                "testCasesCouldNotBeUpdated": "The following test cases could not be updated: {{testCases}}.",
+                "noUnsavedChanges": "No unsaved changes",
+                "unsavedChanges": "Unsaved changes",
+                "updatedTestCasesShort": "Updated test cases",
+                "updatedTestCases": "Warning: You have updated the exercise's test cases, but you have not run the student submissions with the new settings. Do you really want to leave the area?",
+                "updatedTestCasesTooltip": "You have updated the test cases, this does not have direct implications on the student results though. With the button to the right of this symbol you can run the submissions of all students with the updated test case settings.",
```

Comment:
"button to the right" sounds a bit vague and might not be true in the future in case we change the layout. I would propose you reference the button with its title here

Reply:
You are right, this doesn't make sense. I'll change the text as suggested.